model/qti/metadata/ontology/GenericLomOntologyClassificationInjector.php

```diff
@@ -25,11 +24,23 @@
 use oat\taoQtiItem\model\qti\metadata\MetadataInjectionException;
 use oat\taoQtiItem\model\qti\metadata\MetadataInjector;
 use oat\taoQtiItem\model\qti\metadata\MetadataValue;
+use common_Logger;
+use core_kernel_classes_Class;
+use core_kernel_classes_Resource;
+use Psr\Log\LoggerInterface;
 
 class GenericLomOntologyClassificationInjector implements MetadataInjector
 {
     use OntologyAwareTrait;
 
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger = null)
+    {
+        $this->logger = ($logger ?? common_Logger::singleton()->getLogger());
```

Comment:
Is there a way to avoid this static call?

Reply:
It seems `AbstractMetadataService` gets the injectors though the service locator, and right now it does not inject an usable logger for this class (if I make `$logger` non-nullable, the call to the constructor fails, and I've found no easy way to fix it).

Reply:
Logger usage and injection has been removed from `GenericLomOntologyClassificationInjector`, so marking this thread as solved.